libass/ass.c

```diff
@@ -1522,17 +1522,29 @@ ASS_Track *ass_new_track(ASS_Library *library)
 
 int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
 {
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+    bool *features[] = {
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+        [ASS_FEATURE_BIDI_BRACKETS] = &track->parser_priv->bidi_brackets,
+#else
+        [ASS_FEATURE_BIDI_BRACKETS] = NULL,
 #endif
+    };
+
+    if (feature == ASS_FEATURE_INCOMPATIBLE_EXTENSIONS) {
+        for (ASS_Feature i = ASS_FEATURE_INCOMPATIBLE_EXTENSIONS + 1;
+                i < sizeof(features) / sizeof(features[0]); i++) {
+            if (features[i])
+                *features[i] = !!enable;
+        }
         return 0;
-    default:
-        return -1;
     }
+
+    if (feature < 0 || feature > (sizeof(features) / sizeof(features[0])) || !features[feature])
```

Comment:
The middle check should say `>=`.

And you may or may not want to say `< ASS_FEATURE_INCOMPATIBLE_EXTENSIONS` instead of `< 0` to more obviously match the loop above.